tests/test_internet_iface_sel.py

```diff
@@ -0,0 +1,96 @@
+import sys
+import os
+import unittest
+import dbus
+import pdb
+import pyric.pyw as pyw
+
+dir_of_executable = os.path.dirname(__file__)
+path_to_project_root = os.path.abspath(os.path.join(dir_of_executable, '..'))
+sys.path.insert(0, path_to_project_root)
+os.chdir(path_to_project_root)
+
+import wifiphisher.common.interfaces as interfaces
+import wifiphisher.common.constants as constants
+import wifiphisher.pywifiphisher as pywifiphisher
+
+class TestInternetIfaceSelection(unittest.TestCase):
+
+    def setUp(self):
+        bus = dbus.SystemBus()
+        nm_proxy = bus.get_object(constants.NM_APP_PATH, constants.NM_MANAGER_OBJ_PATH)
+        nm = dbus.Interface(nm_proxy, dbus_interface=constants.NM_MANAGER_INTERFACE_PATH)
+        devs = nm.GetDevices()
+        self.internet_iface = ""
+        self.other_ifaces = []
+
+        for dev_obj_path in devs:
+            dev_proxy = bus.get_object(constants.NM_APP_PATH, dev_obj_path)
+            dev = dbus.Interface(dev_proxy, dbus_interface=dbus.PROPERTIES_IFACE)
+            state = dev.Get(constants.NM_DEV_INTERFACE_PATH, 'State')
+            iface_name = dev.Get(constants.NM_DEV_INTERFACE_PATH, 'Interface')
+            #check connected device
+            if state == 100:
+                self.internet_iface = str(iface_name)
+            elif pyw.iswireless(str(iface_name)):
+                self.other_ifaces.append(str(iface_name))
+
+    def sel_ap_iface(self, ap_iface_name):
+        self.nm.set_ap_iface(ap_iface_name)
+        self.nm.check_ifaces_uncontrolled_by_nm()
+
+    def sel_jamming_and_ap_ifaces(self, ap_if_name, jam_if_name):
+        self.nm.set_jam_iface(jam_if_name)
+        self.nm.set_ap_iface(ap_if_name)
+        self.nm.check_ifaces_uncontrolled_by_nm()
+        
+    def test_no_jamming_with_internet(self):
+        """
+        Test with -nJ case with internet connection
+        """
+        if not self.internet_iface:
+            self.fail('Make sure to have internet access')
```

Comment:
Tests should be run without any hardware requirements. You can mock interface objects to simulate a testing environment.